src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,106 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private static final int RESIZING_CONSTANT = 2;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap(int capacity) {
+        table = new Node[capacity];
+        size = 0;
+    }
+
+    public MyHashMap() {
+        this(DEFAULT_CAPACITY);
+    }
 
     @Override
     public void put(K key, V value) {
+        if (size >= table.length * LOAD_FACTOR) {
+            resizeTable();
+        }
+
+        int index = (key == null) ? 0 : getIndex(key, table.length);
+        Node<K, V> newNode = new Node<>(key, value, null);
 
+        if (table[index] == null) {
+            table[index] = newNode;
+        } else {
+            Node<K, V> current = table[index];
+            Node<K, V> prev = null;
+            while (current != null) {
+                if (Objects.equals(current.key, key)) {
+                    current.value = value;
+                    return;
+                }
+                prev = current;
+                current = current.next;
+            }
+            prev.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        int index = (key == null) ? 0 : getIndex(key, table.length);
+        Node<K, V> current = table[index];
+        while (current != null) {
+            if (Objects.equals(current.key, key)) {
+                return current.value;
+            }
+            current = current.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getIndex(K key, int capacity) {
+        return (key == null) ? 0 : (key.hashCode() & (capacity - 1));
```

Comment:
what does & (capacity - 1)) mean? do you understand how it works?

maybe it is better to write implementation by yourself?